app/controllers/users_controller.rb

```diff
@@ -1108,7 +1108,10 @@ def revoke_auth_token
     user = fetch_user_from_params
     guardian.ensure_can_edit!(user)
 
-    if !SiteSetting.log_out_strict && params[:token_id]
+    if params[:token_id]
+      token = UserAuthToken.find_by(id: params[:token_id], user_id: user.id)
+      # The user should not be able to revoke the auth token of current session.
+      raise Discourse::NotFound if guardian.auth_token == token.auth_token
```

Comment:
NotFound is not really the right error message, I would go with a simple 400 here Discourse::InvalidParameters (will change it in a follow up commit)